src/lib/rpcClient.ts

```diff
@@ -14,46 +14,81 @@ import { metrics } from './metric'
 import { config } from './config'
 import { logger } from './logger'
 
+// Type definition for API parameters
+type APIParams = Record<string, unknown>
+
 // Use custom rpc client instead of comet38Client to set keepAlive option
 export class RPCClient {
   private rpcUris: string[]
-  private requestTimeout: number
   private currentIndex = 0
+  private requestTimeout: number
 
   constructor(rpcUri: string | string[]) {
     this.rpcUris = Array.isArray(rpcUri) ? rpcUri : [rpcUri]
     this.requestTimeout = config.rpcRequestTimeout || 5000
   }
 
-  private getRequester(uri: string): APIRequester {
-    return new APIRequester(uri, {
+  private getAxiosInstance(uri: string): AxiosInstance {
+    return axios.create({
+      baseURL: uri,
+      timeout: this.requestTimeout,
       httpAgent: new http.Agent({ keepAlive: true }),
       httpsAgent: new https.Agent({ keepAlive: true }),
-      timeout: this.requestTimeout,
+      headers: {
+        'Content-Type': 'application/json',
+      },
     })
   }
 
   private async request<T>(
     method: 'get' | 'post',
     path: string,
-    params?: APIParams
+    params?: APIParams,
+    jsonRequest?: string | Record<string, unknown>
   ): Promise<{ response: T; uri: string }> {
     let retryCount = 0
+
     while (true) {
       const uri = this.rpcUris[this.currentIndex]
-      const requester = this.getRequester(uri)
-      logger.debug(`[RPC] Making request to ${uri} - ${path}`)
 
       try {
         let response: T
-        if (method === 'post') {
-          response = await requester.post(path, params)
+        const axiosInstance = this.getAxiosInstance(uri)
+
+        if (jsonRequest) {
+          // Handle direct JSON-RPC request
+          logger.debug(`[RPC] Making direct JSON-RPC request to ${uri}`)
+
+          // Parse JSON string to object if needed
+          const requestBody =
+            typeof jsonRequest === 'string'
+              ? (JSON.parse(jsonRequest) as Record<string, unknown>)
+              : jsonRequest
+
+          const axiosResponse = await axiosInstance.post('', requestBody)
+          response = axiosResponse.data as T
```

Comment:
I think we can use the post logic (line 75-76) by passing `jsonRequest` as `Params`